GrosmarkAD lfp class

CodyCBakerPhD opened the pull request:
@bendichter Essentially the same LFP class as used by Yuta and Watson, but with no reference electrodes used.

This will be the last usage of a NoRecording extractor in the Buzsaki conversion set.

bendichter commented on buzsaki_lab_to_nwb/grosmark_code/grosmarknorecording.py:
```diff
@@ -0,0 +1,26 @@
+import spikeextractors as se
+
+
+class GrosmarkNoRecording(BaseRecordingExtractorInterface):
```
Is there something Grosmark-specific about this? Should it just be a `NoRecording`?

CodyCBakerPhD replied:
It will end up being Watson, Yuta, and Grosmark specific since all the subsequent conversions will not use it. But yeah, it could, for those three, be generalized to a non-experiment specific NoRecording Interface if you wanted

bendichter replied:
Yeah, right now this class is copied in Yuta, which isn't good. In general, I'd like to avoid copy/pasted code

CodyCBakerPhD replied:
I'll go ahead and generalize it then really quick.

bendichter commented on buzsaki_lab_to_nwb/grosmark_code/grosmarklfpdatainterface.py:
```diff
@@ -0,0 +1,49 @@
+
+    def convert_data(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False):
+        """Convert the LFP portion of a particular session of the GrosmarkAD dataset."""
+        session_path = self.input_args['folder_path']
+        all_shank_channels = metadata['all_shank_channels']
+        lfp_sampling_rate = metadata['lfp_sampling_rate']
+        spikes_nsamples = metadata['spikes_nsamples']
+        shank_channels = metadata['shank_channels']
+        n_total_channels = metadata['n_total_channels']
```
Why can't you pull these values here?

CodyCBakerPhD replied:
The current format is somewhat legacy to reduce the number of times the .xml has to be read from (not a terribly expensive computation, I'll admit). Basically, all of these fields are used in the metadata of the electrode table so under this design the NWBConverter accesses the .xml once, pulls that info, and passes it to both the RecordingExtractorInterface and the LFPInterface, but with differently keyed metadata since it's still per-interface here.

Once the new changes go through of course, all metadata will occur only once in the total metadata object and the interfaces will simple find it there, or pull what they need within their own interface.

bendichter replied:
@CodyCBakerPhD Can you change it so that this DataInterface pulls this metadata directly from the xml? Is there some pending change that is blocking that?

CodyCBakerPhD replied:
@bendichter Sure, I can change that now.

CodyCBakerPhD replied:
Resolved on this interface

bendichter commented:
@CodyCBakerPhD would it also be possible to remove this from the Converter?

CodyCBakerPhD commented:
@bendichter I will remove those lines of the metadata from the get_metadata() in the converter, yeah. Also the name/description stuff since that's the only thing being passed any more.

CodyCBakerPhD commented:
@bendichter OK, removed all references to specific NoRecording classes in favor of the generalized class. Also removed all lfp metadata from the main converter, the LFPDataInterface fills it all in.
